base_partner_sequence/models/partner.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2004-2009 Tiny SPRL (<http://tiny.be>).
+# Copyright 2013 initOS GmbH & Co. KG (<http://www.initos.com>).
+# Copyright 2016 Tecnativa - Vicent Cubells
+# Copyright 2016 Camptocamp - Akim Juillerat (<http://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models, exceptions, _
+
+
+class ResPartner(models.Model):
+    """Assigns 'ref' from a sequence on creation and copying"""
+
+    _inherit = 'res.partner'
+
+    @api.multi
+    def _get_next_ref(self, vals=None):
+        return self.env['ir.sequence'].next_by_code('res.partner')
+
+    @api.model
+    def create(self, vals):
+        if not vals.get('ref') and self._needsRef(vals=vals):
+            vals['ref'] = self._get_next_ref(vals=vals)
+        return super(ResPartner, self).create(vals)
+
+    @api.multi
+    def copy(self, default=None):
+        default = default or {}
+        if self._needsRef():
+            default['ref'] = self._get_next_ref()
+        return super(ResPartner, self).copy(default)
+
+    @api.multi
+    def write(self, vals):
+        for partner in self:
+            if not vals.get('ref') and partner._needsRef(vals) and \
+               not partner.ref:
+                vals['ref'] = self._get_next_ref(vals=vals)
+            super(ResPartner, partner).write(vals)
+        return True
+
+    @api.multi
+    def _needsRef(self, vals=None):
+        """
+        Checks whether a sequence value should be assigned to a partner's 'ref'
+
+        :param cr: database cursor
+        :param uid: current user id
```

Comment:
Hi, remove from the comments the old reference to cr, uid, id if not used in the code